### ISSUE TITLE
Vagrant virtual machine issues

### ISSUE THREAD

**Issue by fsherratt** [fsherratt/upgraded-parakeet#58]
When running a vagrant based virtual machine on my computer there have been three issues. 
 - The launch hangs during attached SCSI disk. 
 - When I get into the GUI it runs incredibly sluggishly
 - Guest additions doesn't install properly

**Comment by fsherratt** [comment]
Adding `vb.customize ["modifyvm", :id, "--graphicscontroller", "vmsvga"]` seems to help with boot reliability

**Comment by fsherratt** [comment]
`vagrant plugin install vagrant-vbguest` for guest additions

**Comment by Neoxas** [comment]
I have yet to see any of these issues.
What version of virtualbox are you running?

**Comment by Neoxas** [comment]
https://www.serverlab.ca/tutorials/virtualization/how-to-auto-upgrade-virtualbox-guest-additions-with-vagrant/
Would suggest running:
vagrant plugin install vagrant-vbguest

**Comment by fsherratt** [comment]
That did install guest additions but broke the GUI. A bit of googling revealed it's an issue with the graphics controller, it needs to be `vboxvga`

Add the following to the vagrantfile 
```vagrantfile
# Hardware virtualisation
vb.customize ["modifyvm", :id, "--hwvirtex", "on"]
# Set only graphics controller that works
vb.customize ["modifyvm", :id, "--graphicscontroller", "vboxvga"]
# Enable shared clipboard
vb.customize ["modifyvm", :id, "--clipboard", "bidirectional"]
vb.customize ["modifyvm", :id, "--draganddrop", "bidirectional"]
```

It's still a little sluggish but is now usable

**Comment by fsherratt** [comment]
To enable USB 3 passthrough

```vagrantfile
# Enable USB
vb.customize ["modifyvm", :id, "--usb", "on"]
vb.customize ["modifyvm", :id, "--usbxhci", "on"]
```

**Comment by Neoxas** [comment]
Well now vbox guest additions appear to have stopped working. I'll investigate this tomorrow.

**Comment by Neoxas** [comment]
Well turns out that it was just the VBox had resized my window to 600x400.
Changing this in the GUI screen and adding the requested changes appears to have fixed it.

Commit below should address this:
https://github.com/fsherratt/upgraded-parakeet/commit/bbdbd608478875061c7aa3ac3d27a67645480c88

**Comment by fsherratt** [comment]
Re: 120f7625b1d2ab78b36dd4e064e280d78556d793, the reason i hadn't enabled 3D acceleration is because of the error below 

![image](https://user-images.githubusercontent.com/13890746/107350964-f82f1680-6ac1-11eb-9c46-d51b88ee6e65.png)

**Comment by fsherratt** [comment]
Also would be good to increase the CPU count, the default is 2.
`v.cpus = 4`

link: [VBoxManage Customizations](https://www.vagrantup.com/docs/providers/virtualbox/configuration#vboxmanage-customizations)

And `.vagrant` needs to be added to `.gitignore`

**Comment by fsherratt** [comment]
All fixed in pull request #59
